### ISSUE TITLE
yyyy

### ISSUE THREAD

**Issue by Ruthwik333** [HarshitKmr10/qwerty-2#1]
import pandas as pd
import numpy as np
import string
import re
import unicodedata
from sklearn.preprocessing import LabelEncoder
from sklearn.pipeline import Pipeline
from sklearn.feature_extraction.text import TfidfVectorizer
import lightgbm as lgb

LE = LabelEncoder()
df = pd.read_excel("MFA.xlsx")
df['Comments'] = df['Comments'].astype(str)
unique_topics = df['Topic'].unique()
print(unique_topics)
def encoding(text):
    if isinstance(text, str):
        text = unicodedata.normalize("NFKD", text)
    """
    Remove unicoded data
    """
    return text

def remove_URL(text):
    """
    Remove URLs 
    """
    return re.sub(r"https?://\S+|www\.\S+", "", text)

def remove_non_ascii(text):
    """
    Remove non-ASCII characters 
    """
    return re.sub(r'[^\x00-\x7f]', r'', text)

def remove_html(text):
    """
    Remove the html 
    """
    html = re.compile(r"<.*?>|&([a-z0-9]+|#[0-9]{1,6}|#x[0-9a-f]{1,6});")
    return re.sub(html, "", text)

def remove_punct(text):
    """
    Remove the punctuation
    """
    return text.translate(str.maketrans('', '', string.punctuation))

print(df.shape)
print(df.head)
df = df[['Comments', "Sub-Topic", "Topic"]]

df.dropna()
print(df.shape)
df['Clean'] = df['Comments'].apply(remove_URL)
df['Clean'] = df['Clean'].apply(remove_non_ascii)
df['Clean'] = df['Clean'].apply(remove_html)
le = LE.fit(df["Topic"])
df["Topic"] = LE.transform(df["Topic"])
x = df["Clean"]
y = df["Topic"]

import lightgbm as lgb
from sklearn.metrics import accuracy_score as acc

lgb_model = Pipeline((("vect", TfidfVectorizer()), ("clc", lgb.LGBMClassifier(boosting_type='gbdt', objective='multiclass'))))
lgb_model.fit(x, y)

train_acc = acc(lgb_model.predict(x), y)
print("Train Accuracy Topic:- ", train_acc)

topic_names = LE.inverse_transform(np.arange(len(lgb_model.classes_)))

df_latest = pd.read_excel("filtered_cr.xlsx")
print(df_latest.info())

df_latest['Clean'] = df_latest['Meeting Notes'].apply(encoding)
df_latest['Clean'] = df_latest['Clean'].apply(lambda x: x.lower() if isinstance(x, str) else x)

df_latest.dropna(subset=['Clean'], inplace=True)  # Drop rows with NaN values in 'Clean' column
x = df_latest['Clean']
test_probabilities = lgb_model.predict_proba(x)
topic_names = LE.inverse_transform(np.arange(len(lgb_model.classes_)))
result_df = pd.DataFrame(test_probabilities, columns=topic_names)

df_latest["predictedtopic"] = lgb_model.predict(x)
df_latest["predictedtopic"] = LE.inverse_transform(df_latest["predictedtopic"])
df_latest = pd.concat([df_latest, result_df], axis=1)
print(le.classes_)
print(df_latest)
print(df_latest["predictedtopic"])
print(df_latest["Topic"])
correct_predictions = df_latest['predictedtopic'] == df_latest['Topic']
accuracy = correct_predictions.sum() / len(df_latest) * 100
print(accuracy)

df_latest.to_excel("predicted_results.xlsx", index=False)
